lib/utils/github.ts

```diff
@@ -86,10 +86,52 @@ const getPullRequestCommitMessageFromUrl = async (url: string): Promise<string[]
   return (data as Commit[]).map((commit: Commit): string => commit.commit.message);
 };
 
+const getGithubIssueDetails = async (url: string): Promise<any> => {
+  const sessionResponse = await supabase.auth.getSession();
+  const githubToken = sessionResponse?.data.session?.provider_token;
+  const [, , , owner, repoName, , issueNumber] = url.split("/");
+
+  const apiUrl = `https://api.github.com/repos/${owner}/${repoName}/issues/${issueNumber}`;
+
+  const response = await fetch(apiUrl, {
+    headers: {
+      Authorization: `token ${githubToken}`,
+    },
+  });
+  const data = await response.json();
+
+  return {
+    title: data.title,
+    body: data.body,
+  };
+};
+
+const getGithubIssueComments = async (url: string): Promise<any> => {
+  const sessionResponse = await supabase.auth.getSession();
+  const githubToken = sessionResponse?.data.session?.provider_token;
+  const [, , , owner, repoName, , issueNumber] = url.split("/");
+
+  const apiUrl = `https://api.github.com/repos/${owner}/${repoName}/issues/${issueNumber}/comments`;
+
+  const response = await fetch(apiUrl, {
+    headers: {
+      Authorization: `token ${githubToken}`,
+    },
+  });
+  const data = await response.json();
+  const allComments = data.map((comment: any) => comment.body).join(" ");
```

Comment:
Can we use octokit's issues API for this?

https://octokit.github.io/rest.js/v20#issues-get-comment

```js
octokit.rest.issues.getComment({
  owner,
  repo,
  comment_id,
});
```